framework/web/ErrorHandler.php

```diff
@@ -99,7 +99,7 @@ protected function renderException($exception)
                 $response->data = $result;
             }
         } elseif ($response->format === Response::FORMAT_HTML) {
-            if (YII_ENV_TEST || isset($_SERVER['HTTP_X_REQUESTED_WITH']) && $_SERVER['HTTP_X_REQUESTED_WITH'] === 'XMLHttpRequest') {
+            if (YII_ENV_TEST || Yii::$app->request->isAjax) {
```

Comment:
Yes that's why I was wondering if it was okay here to duplicate some of that logic.
On the other hand, the Request object gets created very early, so does the header collection. So they are most likely already in memory.
--> Not sure how to test this.